src/form/AddCardForm.tsx

```diff
@@ -0,0 +1,118 @@
+import Card from "../components/Card.tsx";
+import NumberInput from "../components/NumberInput.tsx";
+import Display from "../components/Display.tsx";
+import Input from "../components/Input.tsx";
+import AddCard from "../components/AddCard.tsx";
+import Modal from "../components/Modal.tsx";
+import {Dispatch, SetStateAction, useState} from "react";
+import HeaderButton from "../components/HeaderButton.tsx";
+import {useStepper} from "../context/StepperContext.tsx";
+
+function AddCardForm() {
+    const {setCurrentStep} = useStepper()
+    const [cardName, setCardName] = useState('')
+    const [name, setName] = useState('')
+    const [expireDate, setExpireDate] = useState({month: '', year: ''})
+    const [cardNumber, setCardNumber] = useState({first: '', second: '', third: '', fourth: ''})
+    const [isModalOpen, setIsModalOpen] = useState(false)
+
+    const handleCardClick = () => {
+        setIsModalOpen(true)
+    }
+
+    const handleCardNameChange = (value: string) => {
+        setCardName(value)
+        setIsModalOpen(false)
+    }
+    const handleInputChange = (value: string, field: string, setState: Dispatch<SetStateAction<any>>) => {
+        setState(prevState => ({
+            ...prevState,
+            [field]: value
+        }))
+    }
+
+    const handleExpiredDateChange = (value: string, field: string) => {
+        handleInputChange(value, field, setExpireDate)
+    }
+
+    const handleCardNumberChange = (value: string, field: string) => {
+        handleInputChange(value, field, setCardNumber)
+    }
+
+    const handleClick = (step: string) => {
+        setCurrentStep(step)
+    }
+
+
+    return (
+        <>
+            <div className="app">
+                <HeaderButton className={"page-title"} tag={"h2"} value={"< 카드 추가"} step={"CardList"} event={() => handleClick("CardList")}/>
```

Comment:
a: (참고) 미션 요구사항인 bottom-up 방식으로 구현한다면 레이아웃 화하여 Header, Button, Icon, Title 컴포넌트로 분리해 [컴파운드 패턴](https://fe-developers.kakaoent.com/2022/220731-composition-component/)으로 구현해 재사용성을 높일 수 있습니다.

```suggestion
                <Button type="button" onClick={() => {
                   setCurrentStep("CardList")
                }}>
                    <Header tag={"h2"} value={"< 카드 추가"} step={"CardList"} event={() => handleClick("CardList")}>
                        <LeftChevronIcon />
                        <Title tag="h2">카드 추가</Title>
                    </Header>
                </Button>
```

Reply:
Bottom up을 장바구니 미션 할 때는 처음부터 한번 해봐야겠네요